plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -650,46 +674,44 @@ func NewServiceAffinityPredicate(podLister algorithm.PodLister, serviceLister al
 // 	the match.
 // Otherwise:
 // 	Create an "implicit selector" which gaurantees pods will land on nodes with similar values
-//	for the affinity labels.
+// 	for the affinity labels.
+//
 // 	To do this, we "reverse engineer" a selector by introspecting existing pods running under the same service+namespace.
 //	These backfilled labels in the selector "L" are defined like so:
 // 		- L is a label that the ServiceAffinity object needs as a matching constraints.
 // 		- L is not defined in the pod itself already.
 // 		- and SOME pod, from a service, in the same namespace, ALREADY scheduled onto a node, has a matching value.
-func (s *ServiceAffinity) CheckServiceAffinity(pod *api.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+
+// WARNING: This Predicate is NOT gauranteed to work if some of the predicateMetadata data isn't precomputed...
+// For that reason it is not exported, i.e. it is highlhy coupled to the implementation of the FitPredicate construction.
+func (s *ServiceAffinity) checkServiceAffinity(pod *api.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+	var services []*api.Service
+	var pods []*api.Pod
+	if pm, ok := meta.(*predicateMetadata); ok && pm.serviceAffinityMatchingPodList != nil && pm.serviceAffinityMatchingPodList != nil {
```

Comment:
@wojtek-t 

You suggested making this bootstrap the services data... so  ive set it up so that at least the unit tests and this can easily be extended to support independently functional `checkServiceAffinity` at a later date.  I'd like to keep as-is for this PR though, b/c 

- Its somewhat noisy to do the query/check the error and is already done elsewhere, and it seems like a generally acceptable thing to expect predicate precomputations to be correct.
- Even if we did precompute serviecs, we also then have to check pods to make semantics consistent, which makes the logic of this method noisy/redundant code.
- If we want  want these functions to be able to query all of their data b/c some is missing, that is probably something we can do more elgantly by leveraging the  precomputations rather than on a case-by-case basis....

if you still feel strongly i can bootstrap the `GetPodServices` function in here, just my 2 cents for now...